### DISCUSSION TITLE
ROS2 intergration

### DISCUSSION THREAD

**Discussion by apgeorg** [ika-rwth-aachen/mqtt_client#11] [Ideas]
Hi there, do you have any plans about a ros2 integration?

**Comment by lreiher** [comment]
Yes, we currently have a WIP ROS2 version of `mqtt_client` for specific message types. We are still working on getting the generic publishers/subscribers to work. We are trying to release a ROS2 version as soon as possible.

**Reply by apgeorg** [reply]
@lreiher thank you!

**Reply by lreiher** [reply]
Note that we now have a ROS2 branch and an almost ready PR #16. The PR is currently blocked by ros/rosdistro#36406. Until the PR in the rosdistro repository is resolved, the package will not be available for installation via apt, but can be built from source.
